aten/src/ATen/core/ATenDispatch.h

```diff
@@ -132,18 +135,97 @@ class CAFFE2_API ATenDispatch {
     return *this;
   }
 
+  ATenDispatch& registerFallbackBoxedOp(TensorTypeId id, FallbackBoxedFunction* fn) {
+    std::lock_guard<std::mutex> lock(mutex_);
+    boxed_fallback_table_[static_cast<size_t>(id)] = fn;
+    return *this;
+  }
+
   const ATenOpTable* getOpTable(const char* schema) const {
     auto iter = op_tables_.find(schema);
     TORCH_CHECK(iter != op_tables_.end(),
         "No functions are registered for schema ", schema);
     return &iter->second;
   }
 
+  const FallbackBoxedFunction* getFallbackBoxedOp(TensorTypeId tid) const {
+    return boxed_fallback_table_[static_cast<size_t>(tid)];
+  }
+
  private:
   std::unordered_map<std::string, ATenOpTable> op_tables_;
+  FallbackBoxedFunction* boxed_fallback_table_[static_cast<int64_t>(TensorTypeId::NumTensorIds)] = {nullptr};
```

Comment:
nit: `std::array` is preferrable to C arrays.

Reply:
You gonna delete this code soon right? I did a C array for consistency with the other array in this file.